src/_modules/experimental/Request.sol

```diff
@@ -204,19 +229,113 @@ library LibRequestClient {
+
+            return self;
+        }
+
+        string[] memory currentValues = self.headers.toJsonObject().getStringArray(key);
+
+        string[] memory newValues = new string[](currentValues.length + values.length);
+
+        for (uint256 i; i < currentValues.length; i++) {
+            newValues[i] = currentValues[i];
+        }
+
+        for (uint256 i; i < values.length; i++) {
+            newValues[i + currentValues.length] = values[i];
+        }
+
+        self.headers.toJsonObject().set(key, newValues);
+
+        return self;
+    }
+
+    function execute(RequestClient memory self, Request memory req) internal returns (ResponseResult) {
+        CommandResult result = toCommand(self, req).run();
+
+        if (result.isError()) {
+            return result.toError().toResponseResult();
+        }
+
+        CommandOutput memory cmdOutput = result.toValue();
+
+        (uint256 status, bytes memory _body, bytes memory _headers) =
+            abi.decode(cmdOutput.stdout, (uint256, bytes, bytes));
+
+        return Ok(
+            Response({
+                url: req.url,
+                status: status,
+                body: _body,
+                headers: jsonModule.create(string(_headers)).unwrap().toResponseHeaders()
+            })
+        );
+    }
+
+    function toCommand(RequestClient memory self, Request memory req) internal returns (Command memory) {
+        string memory curlWriteOutTemplate = "\"\\n%{header_json}\\n\\n%{http_code}\" ";
+
+        if (self._curlVersion.lessThan(semver.create(7, 83))) {
+            curlWriteOutTemplate = "\"\\n{}\\n\\n%{http_code}\" ";
+        }
+
+        string memory script = string.concat(
+            "response=$(curl -s -w ", curlWriteOutTemplate, req.url, " -X ", LibRequest.toString(req.method)
+        );
+
+        string[] memory headersKeys = req.headers.getKeys();
+
+        for (uint256 i; i < headersKeys.length; i++) {
+            string memory key = headersKeys[i];
+            string[] memory headersValues = req.headers.getAll(key);
+
+            for (uint256 j; j < headersValues.length; j++) {
+                script = string.concat(script, " -H ", '"', key, ": ", headersValues[j], '"');
+            }
+        }
+
+        if (req.body.length > 0) {
+            script = string.concat(script, " -d ", "'", string(req.body), "'");
+        }
+
+        script = string.concat(
+            script,
+            ');reverse_response=$(echo "$response" | awk "{a[i++]=\\$0} END {for (j=i-1; j>=0;) print a[j--]}");headers=$(echo "$reverse_response" | awk -v RS="\\n\\n" "NR==2" | awk "{a[i++]=\\$0} END {for (j=i-1; j>=0;) print a[j--]}");code=$(echo "$reverse_response" | awk -v RS="\\n\\n" "NR==1" | awk "{a[i++]=\\$0} END {for (j=i-1; j>=0;) print a[j--]}");body=$(echo "$reverse_response" | awk -v RS="\\n\\n" "NR>2" | awk "{a[i++]=\\$0} END {for (j=i-1; j>=0;) print a[j--]}");cast abi-encode "response(uint256,string,string)" "$code" "$body" "$headers";'
```

Comment:
it would be nice to format this better and/or have a comment with how the script would look once fully built, in its current state it is hard to read

Reply:
this way we can also think more about simplifying it or changing it later